builtin/providers/aws/structure_test.go

```diff
@@ -824,18 +824,46 @@ func TestFlattenResourceRecords(t *testing.T) {
 			Value: aws.String("127.0.0.1"),
 		},
 		&route53.ResourceRecord{
-			Value: aws.String("127.0.0.3"),
+			Value: aws.String("\"abc def\" \"xxx yyy\""),
 		},
 	}
 
-	result := flattenResourceRecords(expanded)
+	expected_unquoted := []string{
+		"127.0.0.1",
+		"abc def\" \"xxx yyy",
+	}
+	expected_verbatim := []string{
+		"127.0.0.1",
+		"abc def\" \"xxx yyy",
+	}
+
+	checkFlattenResourceRecords(t, "TXT", expanded, expected_unquoted)
+	checkFlattenResourceRecords(t, "SPF", expanded, expected_unquoted)
+
+	checkFlattenResourceRecords(t, "CNAME", expanded, expected_verbatim)
+	checkFlattenResourceRecords(t, "MX", expanded, expected_verbatim)
```

Comment:
Ah, a stupid mistake in the test.  Will fix.

And thanks for the tip about subtests.  Will switch to using them.  (I didn't realize that tests failures don't show the whole stack.)